xbmc/settings/Settings.cpp

```diff
@@ -659,6 +660,15 @@ void CSettings::InitializeDefaults()
 
   if (g_application.IsStandAlone())
     std::static_pointer_cast<CSettingInt>(GetSettingsManager()->GetSetting(CSettings::SETTING_POWERMANAGEMENT_SHUTDOWNSTATE))->SetDefault(POWERSTATE_SHUTDOWN);
+
+  // Check if a device UUID is set, and if not, set the default.
+  // Without this check this would cause re-setting it on every boot.
+  // FIXME: This does not work either. Checking for emptiness & using SetValue fails likewise. Ideas?
+  std::shared_ptr<CSettingString> deviceUUID = std::static_pointer_cast<CSettingString>(GetSettingsManager()->GetSetting(CSettings::SETTING_SERVICES_DEVICEUUID));
+  if (deviceUUID->GetValue().empty()) {
+    const std::string& uuid = StringUtils::CreateUUID();
+    std::static_pointer_cast<CSettingString>(GetSettingsManager()->GetSetting(CSettings::SETTING_SERVICES_DEVICEUUID))->SetDefault(uuid);
```

Comment:
don't set the default here. Instead, create and set a new uuid.

Reply:
Great, I'm unsure why it didn't work in the earlier try. I fixed that and force pushed the update in!